Initial set of changes needed for making my application work with the latest cocotron.

cyclaero opened the pull request:
I added separate comments to the individual commits

Commit:
changes for building cocotron-master on Mac OS X 10.0.4 + Xcode 6.4
added directives for debugging the Cocotron frameworks

Commit:
added more files/directories to .gitignore

Commit:
binding doesn't work for none binded NSTextFields -- deactivate bindinginfo until the real issues has been sorted out

Commit:
NSDrawerWindows can become key, otherwise NSTextFields and other input controls are inactive on NSDrawers

Commit:
improve roundDouble()

Commit:
checking whether a Windows is going to close before resizing, resolves a crasher when quitting a Cocotron application
improve dirtyrect handling

Commit:
NSTableHeaders did not show the NSTableColumn titles

Commit:
remove logic for not to draw the NSTableHeader border out of uxtheme class

cyclaero committed:
improved enumeration of locale directories

cyclaero committed:
fixed spelling of Japanese locale

cyclaero committed:
fine adjustment of the drawer height so it matches the available content size on Mac OS X
fixed an issue whereby the content view of the drawer moved upward upon opening-closing-opening the drawer

cjwl commented:
I have started to merge these, I am reviewing the workarounds to get an idea of how to fix the underlying issues.

cyclaero committed:
optimized -[NSTableView drawGridInClipRect:]
corrected vertical position of table header and table corner drawing by NSGraphicsStyle_uxtheme -- was off by 1px to the top
